src/joinUrl.ts

```diff
@@ -0,0 +1,18 @@
+export const joinUrl = (...args: string[]): string => {
+	// Remove any leading or trailing slashes from all parts and then join cleanly on
+	// a single slash, preventing malformed urls
+	const trimmed = args
+		.map((part) => {
+			// Trim left
+			if (part.substr(0, 1) === '/') return part.slice(1);
+			return part;
+		})
+		.map((part) => {
+			// Trim right
+			if (part.substr(part.length - 1, 1) === '/')
+				return part.slice(0, -1);
```

Comment:
```suggestion
			if (part.endsWith('/'))	return part.slice(0, -1);
```

Reply:
No longer needed (but yes, this is much better)